### PR TITLE
TINKERPOP-1664 StarVertexProperty will check meta-properties are valid.

### PR DESCRIPTION
**Pull request by BrynCooke**
StarVertexProperty will check meta-properties are valid.

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -64,6 +64,7 @@ TinkerPop 3.2.5 (Release Date: NOT OFFICIALLY RELEASED YET)
 * Improved error handling of compilation failures for very large or highly parameterized script sent to Gremlin Server.
 * Fixed a bug in `RangeByIsCountStrategy` that changed the meaning of inner traversals.
 * Improved Gremlin-Python Driver implementation by adding a threaded client with basic connection pooling and support for pluggable websocket clients.
+* StarVertexProperty will check meta-properties are valid.
```
Comment (okram) [inline CHANGELOG.asciidoc]: This should be written as:

Fixed a bug in `StarVertexProperty` where properties are now validated.

**Comment by okram** [comment]
VOTE +1

**Comment by robertdale** [comment]
Looks like a candidate for tp31

**Comment by BrynCooke** [comment]
I've reworked for tp31. Also I check null values on array parameters.

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/ElementHelper.java [hunk 2]
```diff
@@ -116,6 +116,10 @@ public static void legalPropertyKeyValueArray(final Object... propertyKeyValues)
         for (int i = 0; i < propertyKeyValues.length; i = i + 2) {
             if (!(propertyKeyValues[i] instanceof String) && !(propertyKeyValues[i] instanceof T))
                 throw Element.Exceptions.providedKeyValuesMustHaveALegalKeyOnEvenIndices();
+
+            if (propertyKeyValues[i + 1] == null) {
```
Comment (okram) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/ElementHelper.java]:
```
if(null == propertyKeyValues[i+1])
  throw Property.Exceptions.propertyValueCanNotBeNull();
```

**Comment by dkuppitz** [comment]
VOTE: +1

And just like the Java compiler I really don't care whether it's `null == propertyKeyValues[i+1]` or `propertyKeyValues[i+1] == null`.

**Comment by spmallette** [comment]
VOTE: +1 - i'll handle the merge

**Pull request merged**